Makefile

```diff
@@ -176,8 +176,8 @@ test:
 	export DYLD_LIBRARY_PATH="${DYLD_LIBRARY_PATH}:${LOCAL_DIR}/lib" && \
 	export LOG_LEVEL=DEBUG && \
 	export RUST_BACKTRACE=1 && \
-	cargo test --no-default-features --features "${ENABLE_FEATURES}" --all ${EXTRA_CARGO_ARGS} -- --nocapture && \
-	cargo test --no-default-features --features "${ENABLE_FEATURES}" --all --bench misc ${EXTRA_CARGO_ARGS} -- --nocapture  && \
+	cargo test --no-default-features --features "${ENABLE_FEATURES}" --all ${EXTRA_CARGO_ARGS} -- --nocapture --test-threads=2 && \
```

Comment:
I don't think it's appropriate. Developers also tend to run tests on a remote high-end server that has a lot of cores.

Reply:
Maybe it can be done only for Mac? Most mac machines only have maybe 4 to 8 threads.

Reply:
so, it is better to check the OS at first and set the thread number?

Reply:
Or check the CPU cores first. And for all machines, use max(1, cores - 2).

Reply:
I don't think the cores is as relevant as the OS - I have 6 cores, but get some failures with more than two threads. I suspect it is due to the relatively low number of FDs that OS X allows.

Anyway, I've update the PR to make this a Mac-only change.